samples: matter: added support for nRF54LC10 (#30038)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/getting_started/hw_requirements.rst

However, it is possible to configure the SoCs with different QSPI or SPI memory if it is supported by Zephyr.
For this purpose, check the reference design for Nordic DKs for information about how to connect the external memory with SoC, specifically whether the pins are designed for the QSPI or the high-speed SPIM operations.

.. note::

>    The nRF54LC10 DK does not come with external memory by default, and it is incapable to fit two slots, required for DFU, in it's internal memory.

Reviewer: ```suggestion
   The nRF54LC10 DK does not come with external memory by default.
   It cannot fit two slots in its internal memory, which is required for DFU.
```

However, it is possible to configure the SoCs with different QSPI or SPI memory if it is supported by Zephyr.
For this purpose, check the reference design for Nordic DKs for information about how to connect the external memory with SoC, specifically whether the pins are designed for the QSPI or the high-speed SPIM operations.

.. note::
   The nRF54LC10 DK does not come with external memory by default.
   It cannot fit two slots in its internal memory, which is required for DFU.

>    Therefore connecting and configuring an external flash is required to perform any type of DFU.

Reviewer: ```suggestion
   Connecting and configuring an external flash is required to perform any type of DFU.
```
